File: cpp/bc1189.cpp
#include <iostream>
#include <iomanip>

using namespace std;

int main()
{
	double cVal[12][12], sum = 0.0;
	int lPos = 5;
	char op;
	
	cin >> op;
	
	for(int i = 0; i < 12; i++)
	{
		for(int j = 0; j < 12; j++)
			cin >> cVal[i][j];
	}
	
	for(int i = 1; i < 6; i++)
	{
		for(int j = 0; j < i; j++)
			sum += cVal[i][j];
	}
	for(int i = 6; i < 11; i++)
	{
		for(int j = 0; j < lPos; j++)
			sum += cVal[i][j];
		
		lPos--;
	}
	
	if(op != 'S') sum /= 30.0;
	
	cout << fixed << setprecision(1) << sum << endl;
}

File: python/bc1189.py
ma = [[None for j in range(12)] for i in range(12)]
sum = 0.0
lPos = 5

op = input()

for c in range(0, 12):
    for d in range(0, 12):
        ma[c][d] = float(input())

for i in range(1, 6):
    for j in range(0, i):
        sum += ma[i][j]

for i in range(6, 11):
    for j in range(0, lPos):
        sum += ma[i][j]
    lPos -= 1

if op != "S": sum /= 30.0

print(f"{sum:.1f}")